src/audio/selector.c

```diff
@@ -210,15 +227,33 @@ static int selector_ctrl_set_cmd(struct comp_dev *dev,
  * \param[in] size Command data size.
  * \return Error code.
  */
-static int selector_ctrl_get_cmd(struct comp_dev *dev,
+static int selector_ctrl_get_data(struct comp_dev *dev,
 				 struct sof_ipc_ctrl_data *cdata, int size)
 {
-	/* TODO add implementation aligned with SW assumptions */
-	cdata->cmd = SOF_CTRL_CMD_ENUM;
-	trace_selector("selector_ctrl_get_cmd(), cdata->cmd = %u",
-		       cdata->index);
+	struct comp_data *cd = comp_get_drvdata(dev);
 
-	return 0;
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		trace_selector("selector_ctrl_get_data(), SOF_CTRL_CMD_BINARY");
+
+		/* Copy back to user space */
+		if (memcpy_s(cdata->data->data, ((struct sof_abi_hdr *)
```

Comment:
This doesn't follow coding convention and it's possible checkpatch doesn't detect it either. it should be 
```
ret = memcpy();
if (ret < 0)
    return ret;
```

Reply:
fixed